click_odoo_contrib/_backup.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018 ACSONE SA/NV.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+import os
+import shutil
+import tempfile
+import zipfile
+from contextlib import contextmanager
+
+
+class AbstractBackup(object):
+    """Abstract class with methods to open, read, write, close,
+    a backup
+    """
+
+    def __init__(self, path, mode):
+        """
+        :param path: Either the path to the file, or a file-like object, or a folder
+                     or ... .
+        :param mode: The mode can be either read "r", write "w" or append "a"
+        """
+        self._path = path
+        self._mode = mode
+
+    def copytree(self, src):
+        """Recursively copy a directory tree to the backup.
+        :param dirname: Directory to copy from
+        Dirname of src dir is the root of copied files into the archive
+        """
+
+    def addfile(self, filename, arcname):
+        """Add a file to the backup.
+
+        :param filename: the path to the souce file
+        :param arcname: the path into the backup
+        """
```

Comment:
I sense a lack of symmetry between addfile and copytree: they could be both named "addsomething", and I'm wondering why one names the destination and the other not. (still nitpicking I know :)

Reply:
indeed.. I'll rework this part. The question is: do we need generic methods? We should may be name the methods `addFilestore`, `writeDbDump`, `writeManifest`?